cmd/gameserver/packet/inventory.go

```diff
@@ -0,0 +1,130 @@
+package packet
+
+import (
+	"github.com/ubis/Freya/share/log"
+	"github.com/ubis/Freya/share/models/character"
+	"github.com/ubis/Freya/share/network"
+	"github.com/ubis/Freya/share/rpc"
+)
+
+func SendStorageExchangeMoveStatusPacket(session *network.Session, result byte) {
+	packet := network.NewWriter(STORAGE_EXCHANGE_MOVE)
+	packet.WriteByte(result)
+	packet.WriteInt32(0)
+
+	session.Send(packet)
+}
+
+func SendStorageExchangeMoveMasterServerPacket(packet string, id int32, deleteSlot uint16, createSlot uint16) character.ItemMoveRes {
+	send := character.ItemMoveReq{Server: byte(g_ServerSettings.ServerId), Id: id, DeleteSlot: deleteSlot, CreateSlot: createSlot}
+	receive := character.ItemMoveRes{}
+	err := g_RPCHandler.Call(packet, send, &receive)
+	if err != nil {
+		log.Error(err.Error())
+	}
+
+	return receive
+}
+
+func StorageExchangeMove(session *network.Session, reader *network.Reader) {
+	equip := reader.ReadUint32()
+	deleteSlot := uint16(reader.ReadUint32())
+	inventory := reader.ReadUint32()
+	createSlot := uint16(reader.ReadUint32())
+
+	ctx, err := parseSessionContext(session)
+	if err != nil {
+		log.Error(err.Error())
+		SendStorageExchangeMoveStatusPacket(session, 0x00)
+		return
+	}
+	playerChar := ctx.char
+
+	isEquip := equip == 1
+	isInventory := inventory == 1
+
+	if isEquip && !isInventory {
+		result := SendStorageExchangeMoveMasterServerPacket(rpc.UnEquipItem, playerChar.Id, deleteSlot, createSlot)
```

Comment:
`playerChar.Id` is not thread-safe, you might want to use `id, err := getSessionCharId(session)`

and everything what modifies or reads from `playerChar` is not thread safe, you need to use lock zone which is in the context, so for example:

```Go
ctx, err := parseSessionContext(session)
// handle err

ctx.mutex.Lock()
ctx.char.Equipment.Remove(slot)
ctx.mutex.Unlock()
```